plugins/parsers/json_v2/parser.go

```diff
@@ -700,15 +700,11 @@ func (p *Parser) convertType(input gjson.Result, desiredType string, name string
 	return input.Value(), nil
 }
 
+// Check if gjson result exists and if a non-existing gjson result can be skipped over
```

Comment:
```suggestion
// Check if gjson result exists and if a non-existing gjson result can be skipped over.
// We need to return an error even for optional entities to allow the upstream code
// to take the skipping return value into account.
```

Reply:
That makes sense!

Another option is to change ``checkResult()`` into a single error return on ``result.Exists()``. Currently, the other arguments are used either in a log message or just passed back in the return - not a ton of checking done. This could leave ``checkResult()`` to be expanded with other types of error messages in the future, but leave evaluating what to do about an error (skip, error out) to the calling logic.

Calls could be changed to something like:

```
if err := p.checkResult(result, c.Path); err != nil {
	if c.Optional {
		continue
	}
	return nil, err
}
(...)
func (p *Parser) checkResult(result gjson.Result, path string) (error) {
	if !result.Exists() {
		return fmt.Errorf("the path %q doesn't exist", path)
	}

}
```

Adding comment or code change both sound good to me.

Reply:
Yeah that would be the better option or even condense your code to
```golang
if !result.Exists() {
	if c.Optional {
		continue
	}
	return nil, fmt.Errorf("the path %q doesn't exist", c.Path)
}
```

Reply:
Just 4 spots, moved the ``optional`` arg to replace ``skip`` and flattened ``checkResult()``